### ISSUE TITLE
[Bug]: Application requesting Location Permission on launch time.

### ISSUE THREAD

**Issue by mahipalsingh-syt** [rnmapbox/maps#2583]
### Mapbox Implementation

Mapbox

### Mapbox Version

~> 10.10.0

### Platform

iOS

### `@rnmapbox/maps` version

10.0.0-beta.65

### Standalone component to reproduce

```
import MapboxGL from "@rnmapbox/maps"

<MapboxGL.MapView style={styles.map} />
```

### Expected behavior

application should only request permission when user tap on location button or  if Mapbox use <MapboxGL.UserLocation/>

### Additional links and references
![Simulator Screen Shot - iPhone 13 - 2023-02-03 at 21 48 02](https://user-images.githubusercontent.com/64964760/216653343-1fb00571-aa42-4be3-bc8b-6c4b3679432a.png)

_No response_

**Comment by harry-isaac** [comment]
I'm seeing this as well

**Comment by mfazekas** [comment]
Fixed by:  #2586
